server/models/annotation.py

```diff
@@ -0,0 +1,534 @@
+        if properyName != 'type'
+    }
+    baseRectangleShapePatternProperties.update(baseShapePatternProperties)
+    rectangleShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectangle']
+                    },
+                },
+                'required': ['type'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False
+            }
+        ]
+    }
+    rectangleGridShapeSchema = {
+        'allOf': [
+            baseRectangleShapeSchema,
+            {
+                'type': 'object',
+                'properties': {
+                    'type': {
+                        'type': 'string',
+                        'enum': ['rectanglegrid']
+                    },
+                    'widthSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                    'heightSubdivisions': {
+                        'type': 'integer',
+                        'minimum': 1
+                    },
+                },
+                'required': ['type', 'widthSubdivisions', 'heightSubdivisions'],
+                'patternProperties': baseRectangleShapePatternProperties,
+                'additionalProperties': False,
+            }
+        ]
+    }
+
+    annotationSchema = {
+        '$schema': 'http://json-schema.org/schema#',
+        'id': '/girder/plugins/large_image/models/annotation',
+        'type': 'object',
+        'properties': {
+            'name': {
+                'type': 'string',
+                # TODO: Disallow empty?
+                'minLength': 1,
+            },
+            'description': {'type': 'string'},
+            'attributes': {
+                'type': 'object',
+                'additionalProperties': True,
+                'title': 'Image Attributes',
+                'description': 'Subjective things that apply to the entire '
+                               'image.'
+            },
+            'elements': {
+                'type': 'array',
+                'items': {
+                    # Shape subtypes are mutually exclusive, so for
+                    #  efficiency. don't use 'oneOf'
+                    'anyOf': [
+                        # If we include the baseShapeSchema, then shapes that
+                        # are as-yet invented can be included.
+                        # baseShapeSchema,
+                        arrowShapeSchema,
+                        circleShapeSchema,
+                        pointShapeSchema,
+                        polylineShapeSchema,
+                        rectangleShapeSchema,
+                        rectangleGridShapeSchema,
+                    ]
```

Comment:
No problem.  We can add any additional shapes as needed.  Would it be the same form as the rectangle?  That is, we give a center, width, height, rotation, and, optionally for 3-D scenarios, a normal?  Or would it be more useful to have it be specified via foci?